### PR TITLE
Modus Chip - Added `max-width` to handle text overflow

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

Test it by adding max-width property to the chip and the ellipsis are enabled when the value reaches end of the chip.

References #
#1771 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

Storybook page
Go to the canvas section modify the max width property by reducing it to 20px from the default set 100px , the default value 'Bryan' will have ellipsis added at the end

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

#1771  is being fixed in this branch;
Test it by adding max-width property to the chip and the ellipsis are enabled when the value reaches end of the chip.

### CODE REVIEW

**Comment by coliff** [comment]
Hello - a quick few points:
- Please always run `npm run lint-css` if you edit any of the SCSS files. It will re-sort all the properties alphabetically.
- I don't think that `storybook/src/index.html` changed in this PR is intended to be used for tests like that.
- Should the Modus chip container include a max-width?

**Comment by ElishaSamPeterPrabhu** [comment]
> Hello - a quick few points:
> - Please always run `npm run lint-css` if you edit any of the SCSS files. It will re-sort all the properties alphabetically.
> - I don't think that `storybook/src/index.html` changed in this PR is intended to be used for tests like that.
> - Should the Modus chip container include a max-width?

Noted @coliff , the max-eidth was set to try whether the chip ellipsis is getting triggered.

**Comment by msankaran0712** [comment]
> Hello - a quick few points:
> 
> * Please always run `npm run lint-css` if you edit any of the SCSS files. It will re-sort all the properties alphabetically.
> * I don't think that `storybook/src/index.html` changed in this PR is intended to be used for tests like that.
> * Should the Modus chip container include a max-width?

@coliff Yes, we need a new configuration `maxWidth`. And it looks like the close icon size is too big 
Modus Chip Web
Modus Styleguide
Modus Figma Web

File: stencil-workspace/src/components/modus-chip/modus-chip.scss [hunk 1]
```diff
@@ -8,6 +8,7 @@
   color: $modus-chip-color;
   cursor: pointer;
   display: inline-flex;
+  max-width: 2px;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.scss]: Please revert this line

File: stencil-workspace/src/components/modus-chip/modus-chip.scss [hunk 2]
```diff
@@ -23,6 +24,8 @@
     height: $rem-16px;
 
     span {
+      background-color: aquamarine;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.scss:27]: Please remove the background color. Replace it with the below highlighted code,

image

File: stencil-workspace/src/components/modus-chip/modus-chip.e2e.ts [hunk 3]
```diff
@@ -132,6 +132,23 @@ describe('modus-chip', () => {
...
+
+    await page.setContent('<modus-chip></modus-chip>');
+    const chip = await page.find('modus-chip');
+    expect(await chip.getProperty('maxWidth')).toBeFalsy();
+
+    chip.setProperty('maxWidth', '100px'); // Set the max-width property
+    await page.waitForChanges();
+    expect(await chip.getProperty('maxWidth')).toEqual('100px');
+
+    const shadowContainer = await page.find('modus-chip >>> .modus-chip');
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.e2e.ts:146]: The CSS selector should now be changed to `modus-chip >>> span`

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx [hunk 4]
```diff
@@ -95,14 +98,16 @@ export class ModusChip {
       ${!this.showCheckmark && !this.imageUrl ? 'no-left-icon' : null}
       ${!this.showClose ? 'no-right-icon' : null}
     `;
-
+    console.log(this.maxWidth);
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.tsx]: Please remove the console log code

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx [hunk 5]
```diff
@@ -95,14 +98,16 @@ export class ModusChip {
...
+    const style = { 'max-width': this.maxWidth ?? undefined };
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.tsx:101]: Please revert your changes and use the below code,

``` js
const style = {
      style: {
        'max-width': this.maxWidth ?? undefined,
      },
    };
    return (
      <div
        aria-disabled={this.disabled ? 'true' : undefined}
        aria-label={this.ariaLabel}
        class={chipClass}
        onClick={this.disabled ? null : (event) => this.onChipClick(event)}
        tabIndex={0}>
        {this.imageUrl ? (
          <img src={this.imageUrl} alt="" />
        ) : this.showCheckmark ? (
          <IconCheck size={'24'}></IconCheck>
        ) : null}
        <span {...style}>{this.value}</span>
        {this.showClose ? (
          <IconRemove onClick={this.disabled ? null : (event) => this.onCloseClick(event)} size={'24'}></IconRemove>
        ) : null}
      </div>
    );
```

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx [hunk 6]
```diff
@@ -36,6 +36,9 @@ export class ModusChip {
   /** (optional) The chip's value. */
   @Prop() value: string;
 
+  /** (optional) The Chip's div max-width */
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.tsx]: Please rewrite the description to `Maximum width for the Chip's text and shows ellipsis when truncated.`

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx [hunk 7]
```diff
@@ -79,6 +79,13 @@ export default {
         type: { summary: 'string' },
       },
     },
+    maxWidth: {
+      description: "The chip's div max-width",
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx]: Please update the description to the latest

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx [hunk 8]
```diff
@@ -79,6 +79,13 @@ export default {
...
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '100px' },
+      },
+    }
   },
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx:89]: Please pass the maxWidth value to the `Default` and `Outline` templates too

File: stencil-workspace/src/index.html [hunk 9]
```diff
@@ -17,5 +33,18 @@
...
+    <!-- <div class="resizable-window" style=" position: fixed;
+    top: 0;
+    left: 0;
+    background-color: grey;
+    padding: 10px;"><modus-chip value="sdasdasdsd"></modus-chip></div> -->
+    <div class="container">
+      <div class="column">Column 2</div>
+      <div class="column">Column 3</div>
+    </div>
+    <div class="column"><modus-chip  value="sdasdahbhugugyugygyugyugyuguyghbhihiuhuihuihuihuihullihuihsdxxxasassssssssd"></modus-chip></div>
```
Comment (msankaran0712) [inline stencil-workspace/src/index.html:45]: Please add some meaningful sentence

File: stencil-workspace/src/index.html [hunk 10]
```diff
@@ -7,7 +7,23 @@
```
Comment (msankaran0712) [inline stencil-workspace/src/index.html:7]: Please revert the changes on this file

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.tsx]: Removed

File: stencil-workspace/src/components/modus-chip/modus-chip.e2e.ts (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.e2e.ts:146]: modified

File: stencil-workspace/src/components/modus-chip/modus-chip.scss (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.scss:27]: Modified

File: stencil-workspace/src/components/modus-chip/modus-chip.scss (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.scss]: changed

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.tsx:101]: Reverted

File: stencil-workspace/src/components/modus-chip/modus-chip.tsx (hunk 6 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.tsx]: Rewritten

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx (hunk 7 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx]: Updated

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx (hunk 8 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx:89]: Added

File: stencil-workspace/src/index.html (hunk 10 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/index.html:7]: The text is not present now

Comment (msankaran0712) [reply stencil-workspace/src/index.html:7]: @ElishaSamPeterPrabhu please copy and paste the code for this file from the main branch.

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx [hunk 11]
```diff
@@ -79,6 +79,13 @@ export default {
         type: { summary: 'string' },
       },
     },
+    maxWidth: {
+      description: "The chip's maximum width",
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx:83]: description can be edited "Chip text's maximum width"

**Comment by msankaran0712** [comment]
> > Hello - a quick few points:
> > 
> > * Please always run `npm run lint-css` if you edit any of the SCSS files. It will re-sort all the properties alphabetically.
> > * I don't think that `storybook/src/index.html` changed in this PR is intended to be used for tests like that.
> > * Should the Modus chip container include a max-width?
> 
> @coliff Yes, we need a new configuration `maxWidth`. And it looks like the close icon size is too big Modus Chip Web Modus Styleguide Modus Figma Web

@coliff  Looks like on production the close icon is big, we have created a separate issue.

**Comment by msankaran0712** [comment]
@ElishaSamPeterPrabhu Please reference the original issue link and update the PR description about how to test the fix.

File: stencil-workspace/src/components/modus-chip/modus-chip.scss [hunk 12]
```diff
@@ -23,7 +23,11 @@
     height: $rem-16px;
 
     span {
+      display: inline-block;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.scss:26]: Please add these rules to the line 46. Currently, the rules are added within a specific css selector, so the chip is not showing the ellipsis.

Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-chip/modus-chip.scss:26]: Added now

File: stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx (hunk 11 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-chip/modus-chip.stories.tsx:83]: Changed

File: stencil-workspace/src/index.html (hunk 10 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/index.html]: I did and committed it separately

**Comment by ElishaSamPeterPrabhu** [comment]
#1771 is being fixed in this branch; 
**Test** it by adding **max-width** property to the chip and the ellipsis are enabled when the value reaches end of the chip.

File: stencil-workspace/src/components/modus-chip/modus-chip.scss [hunk 13]
```diff
@@ -23,7 +23,11 @@
...
       font-size: $rem-12px;
+      overflow: hidden;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-chip/modus-chip.scss:28]: Kindly remove the new rules from this selector as we have it already added to line 50 which is nested directly under the component so it applies all the time.

**Review by msankaran0712 — APPROVED** [review]
